internal/resolver/delegatingresolver/delegatingresolver_test.go

```diff
@@ -0,0 +1,71 @@
+/*
+ *
+ * Copyright 2018 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package delegatingresolver
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/internal/testutils"
+	"google.golang.org/grpc/resolver"
+	"google.golang.org/grpc/resolver/manual"
+	"google.golang.org/grpc/serviceconfig"
+)
+
+// TestDelegatingResolverNoProxy verifies the behavior of the delegating resolver when no proxy is configured.
+func TestDelegatingResolverNoProxy(t *testing.T) {
+	t.Setenv("HTTPS_PROXY", "")               // Explicitly set proxy environment to empty to mimic no proxy environment set
+	mr := manual.NewBuilderWithScheme("test") // Set up a manual resolver to control the address resolution.
+	target := "test:///localhost:1234"
+
+	stateCh := make(chan resolver.State, 1)
+	updateStateF := func(s resolver.State) error {
+		select {
+		case stateCh <- s:
+		default:
+		}
```

Comment:
Why do you have a `default` arm in these `select` statements? We don't want to miss any updates/errors happening as part of the test.

If you are concerned about a bug which could lead to these `select`s hanging forever, you could instead create a test context with a default test timeout, and have an arm in these `select` statements for `<-ctx.Done()`. That will ensure that if there is a failure in the test, these will not hang forever.